app/controllers/api/v1/apps_controller.rb

```diff
@@ -2,6 +2,9 @@
 
 class Api::V1::AppsController < Api::BaseController
   skip_before_action :require_authenticated_user!
+  before_action :validate_token_endpoint_auth_method!
+
+  TOKEN_ENDPOINT_AUTH_METHODS = %w(none client_secret_basic client_secret_post).freeze
```

Comment:
We'll actually be able to fetch these from Doorkeeper in the future: https://github.com/doorkeeper-gem/doorkeeper/pull/1772

Essentially `Doorkeeper.config.client_authentication_methods.map(&:name)`